### PR TITLE
Add vector storage option.

### PR DESCRIPTION
**Pull request by astamm**

### CODE REVIEW

File: R/nloptions.R [hunk 1]
```diff
@@ -35,36 +35,31 @@
...
-    if (!all(names(optlist) %in% names(opts)))
-        warning("Unknown names in control: ",
-                names(optlist)[!(names(optlist) %in% names(opts))])
+  for (option_name in names(optlist))
+    opts[[option_name]] <- optlist[[option_name]]
 
-    if (!is.null(namc))
-        opts[namc] <- optlist[namc]
+  if ("algorithm" %in% names(opts)) {
+    cli::cli_alert_warning(
```
Comment (eddelbuettel) [inline R/nloptions.R]: I don't think you import or even suggests `cli` yet.  On the margin, could it not be base R's `warning()` here as before?

Comment (astamm) [reply R/nloptions.R]: My bad, I was fixing it while you were reviewing.
A commit is on its way. Thanks

Comment (astamm) [reply R/nloptions.R]: I am though thinking about replacing calls to `warning()`, `stop()` and `cat()` to use a more elegant command-line interface as provided by **cli**. But it's not urgent and purely aesthetics.

Comment (eddelbuettel) [reply R/nloptions.R]: Sure, your call.  And these days the penny dropped in most places and many of these convenience packages have a low or zero count for _hard_ dependencies and moved the long tail of things they may bring in to Suggests.  `glue` is similarly popular with a narrow hard dependency list.

**Pull request merged**